### PR TITLE
DF-551 website long filter search

### PR DESCRIPTION
**Pull request by TNA-Allan**
Ticket URL: https://national-archives.atlassian.net/browse/DF-551

## About these changes

These changes enables `See more` for Website search and allows interaction with the long filter. 
Currently its the `topic` field of Website search which shows `See more`

The long filter methods in `CatalogueSearchLongFilterView` are moved to `BaseLongFilteredSearchView` so that Catalogue search for long filter and Website search for long filter (new) uses the same.

Fixes show no results for a refine query for a non matching criteria.

## How to check these changes

As mentioned in the ticket

## Before assigning to reviewer, please make sure you have

- [x] Checked things thoroughly before handing over to reviewer.
- [x] Checked PR title starts with ticket number as per project conventions to help us keep track of changes.
- [x] Ensured that PR includes only commits relevant to the ticket.
- [x] Waited for all CI jobs to pass before requesting a review.
- [x] Added/updated tests and documentation where relevant.

## Merging a feature or bug fix?

1. Use the `Squash and merge` option to keep the target branch's commit history nice and clean.
2. Where relevant, include the ticket number in commit title, e.g. `DF-XXX: Ticket name / short description`.

## Merging a release branch?

1. Use the `Create a merge commit` option to preserve the original commit IDs.
2. Use the message format `release/<major.minor.patch>` when merging.

## Deployment guidance

1. Post to the appropriate Slack channel to check that it is okay to continue.
2. To start a deployment, push the relevant branch to the `platform` remote.
3. Update the appropriate Slack channel when the deployment is complete.
4. Update the status on the corresponding Jira ticket (where relevant).

### CODE REVIEW

**Comment by ababic** [comment]
@TNA-Allan I'm afraid this is going to have massive amounts of conflict with #729. We need to decide which should be merged first :)

**Comment by TNA-Allan** [comment]
> @TNA-Allan I'm afraid this is going to have massive amounts of conflict with #729. We need to decide which should be merged first :) It looks like the only changes to `website_search.html` and `catalogue_search.html` here might be superficial only, so I would suggest #729 comes first, and the changes here be reverted. Thoughts?

@ababic  converted to Draft

**Comment by TNA-Allan** [comment]
Hi @ababic , following merges (from https://github.com/nationalarchives/ds-wagtail/pull/729) , submitting for your review. Thanks.

File: templates/search/website_search_long_filter_chooser.html [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+            <div class="long-filters__submit-section">
+                <input type="submit" value="Apply filters" class="search-filters__submit">
+            </div>
+        </form>
+    </div>
+{% endblock %}
+
+{% block extra_js %}
+    <script src="{% static 'scripts/catalogue_search.js' %}"></script>
+{% endblock %}
```
Comment (ababic) [inline templates/search/website_search_long_filter_chooser.html:42]: Question: Is there a difference between this template and the one already used for the 'catalogue' version? 

Suggestion: If the interfaces do not need to differ, having a single shared template will be easier to maintain.

Comment (ababic) [reply templates/search/website_search_long_filter_chooser.html:42]: NOTE: Maybe changing the existing template name and adding `template="search/long_filter_options.html"` `BaseLongFilterOptionsView.template_name`

File: etna/search/views.py [hunk 2]
```diff
@@ -580,22 +580,11 @@ def get_datalayer_data(self, request: HttpRequest) -> Dict[str, Any]:
...
-    search_tab = SearchTabs.CATALOGUE.value
-
-
-class CatalogueSearchLongFilterView(BaseFilteredSearchView):
-    api_method_name = "search"
-    api_stream = Stream.EVIDENTIAL
-    default_group = "tna"
-    form_class = CatalogueSearchForm
-    template_name = "search/catalogue_search_long_filter_chooser.html"
+class BaseLongFilteredSearchView(BaseFilteredSearchView):
```
Comment (ababic) [inline etna/search/views.py:583]: Optional: This naming is not bad, but I would suggest ditching 'Search' from the name (that's not really it's 'function') and try to better clarify what the 'function' of the view is. How do you feel about `BaseLongFilterOptionsView`?

File: templates/search/website_search_long_filter_chooser.html (hunk 1 above)
Comment (ababic) [reply templates/search/website_search_long_filter_chooser.html:42]: `{% url 'search-website' %}` / `{% url 'search-catalogue' %}` feels like exactly the sort of thing that could be added to the context by the respective views.

Comment (TNA-Allan) [reply templates/search/website_search_long_filter_chooser.html]: Good spot and thanks for the suggestions @ababic

File: etna/search/views.py (hunk 2 above)
Comment (TNA-Allan) [reply etna/search/views.py]: Thanks and agreed on `BaseLongFilterOptionsView` . Also looking with the template `long_filter_options.html` seems clearer.

**Review by ababic — APPROVED** [review]
Thanks @TNA-Allan. Happy to approve this. Just caught a couple of **very** minor things.

File: templates/search/catalogue_search.html [hunk 3]
```diff
@@ -27,49 +27,50 @@
...
+                <div>
+                    {{ form.group.errors }}
+                    {{ form.group.as_hidden }}
+                </div>
 
+            {% endif %}
         {% endif %}
-    {% endif %}
+
+    {% endwith  %}
```
[the remark this answers is not in the document]
Comment (TNA-Allan) [reply templates/search/catalogue_search.html:72]: Fixed

**Pull request merged**